### DISCUSSION TITLE
Is it possible to fix a file at a time rather than project/sln?

### DISCUSSION THREAD

**Discussion by caveman-dick** [JosefPihrt/Roslynator#1029] [Q&A] [answered]
I've been implementing linters and formatters to run under a commit-hook using [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged) so that it will fix the files in the staging area and nothing else.

Is it possible to run the fix command so that it only works on a single file at a time rather than all the files? e.g.

```
roslynator fix --supported-diagnostics RCS1036 ./x/y/Startup.cs
```

**Comment by caveman-dick** [comment]
Not the answer, but I've managed to use a workaround using `dotnet format` to run the analyzer: 

```bash
dotnet format Solution1.sln analyzers --diagnostics RCS1036 --severity info --include ./x/y/Startup.cs
```

**Comment by josefpihrt** [comment, marked as answer]
Hi,

Roslynator command line tool currently does not provide functionality to filter by file.

You are right that `dotnet format` can be used for this purpose.

**Comment by josefpihrt** [comment]
https://github.com/JosefPihrt/Roslynator/issues/1030
